superset/sql_lab.py

```diff
@@ -205,7 +206,8 @@ def execute_sql_statement(
     ):
         if SQL_MAX_ROW and (not query.limit or query.limit > SQL_MAX_ROW):
             query.limit = SQL_MAX_ROW
-        if query.limit:
+        # do not apply limit for databases that don't support LIMIT in the query
+        if query.limit and db_engine_spec.limit_method != LimitMethod.FETCH_MANY:
```

Comment:
Should we change this to base db_engine_spec instead?

Reply:
Sorry, what do you mean?

Reply:
@dpgaspar we actually don't need this, in `apply_limit_to_sql` it already keeps the query unmodified if the method if `FETCH_MANY`.